x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/Types.java

```diff
@@ -91,7 +91,8 @@ private static void walkMapping(String name, Object value, Map<String, EsField>
                     break;
                 case UNSUPPORTED:
                     String type = content.get("type").toString();
-                    field = new UnsupportedEsField(name, type);
+                    field = new UnsupportedEsField(name, type, null, properties);
+                    propagateUnsupportedType(name, type, properties);
```

Comment:
Why is propagated needed? The parents should always be discovered before children which means an unsupported parent will discovered before its children, which in turn, will already know their parent is unsupported and thus become unsupported as well.

Reply:
@costin  If I remember well, the method pair `startWalking`-`walkMapping` does the discovery the other way around: start with a mapping looking like `parent=[child=[another_child=[type:keyword]], type:text], type:foobar]]`, get the parent, look at properties like "it's nested", or "it's object", or "has sub-fields" [here](https://github.com/elastic/elasticsearch/blob/a9e92964d51ee46440627913aa6fb3aca81401c8/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/Types.java#L65-L77), if yes, dig into those and so on and so forth.

Then after discovering all the children, create the parent itself [here](https://github.com/elastic/elasticsearch/blob/a9e92964d51ee46440627913aa6fb3aca81401c8/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/Types.java#L92-L96). So, by the time the algorithm finds out that a root field is unsupported, the children already have some types (their own).

Reply:
This approach is used only in the case of some tests that use a `.json` file with mappings. I made this change because I wanted to also test the error message ([here](https://github.com/elastic/elasticsearch/pull/50823/files#diff-334bd2f8c33b60789817181fcef4ca52R503)) that gets printed out from the Analyzer.